### ISSUE TITLE
Implement Minimal Training Loop\

### ISSUE THREAD

**Issue by Bernard897** [Bernard897/deca-NLP#13]
Design and implement a minimal training loop, including forward pass, loss calculation, and backpropagation. Implement a mechanism to track validation loss.\
